### PR TITLE
Add support for debug rendering obbs

### PR DESCRIPTION
**Pull request by slimbuck**
Add optional matrix so debug AABB rendering can be oriented.

### CODE REVIEW

**Comment by mvaligursky** [comment]
We discussed this here: https://github.com/playcanvas/engine/discussions/3073#discussioncomment-590090
I was keen to add it back then, but it was agreed to keep those outside of the code engine. Perhaps this can be still outside?

**Comment by mvaligursky** [comment]
Lets add it, and create a ticket to move all these non-pubic rendering helpers to extras for now?

File: src/framework/app-base.js [hunk 1]
```diff
@@ -1794,15 +1794,16 @@ class AppBase extends EventHandler {
...
      * @example
      * // Render a red wire aligned box
      * const min = new pc.Vec3(-1, -1, -1);
      * const max = new pc.Vec3(1, 1, 1);
      * app.drawWireAlignedBox(min, max, pc.Color.RED);
      * @ignore
      */
-    drawWireAlignedBox(minPoint, maxPoint, color = Color.WHITE, depthTest = true, layer = this.scene.defaultDrawLayer) {
-        this.scene.immediate.drawWireAlignedBox(minPoint, maxPoint, color, depthTest, layer);
+    drawWireAlignedBox(minPoint, maxPoint, color = Color.WHITE, depthTest = true, layer = this.scene.defaultDrawLayer, mat = undefined) {
```
Comment (willeastcott) [inline src/framework/app-base.js:1805]: Why do you need the `= undefined` here?

**Comment by mvaligursky** [comment]
Created an issue: https://github.com/playcanvas/engine/issues/6024

File: src/framework/app-base.js (hunk 1 above)
Comment (slimbuck) [reply src/framework/app-base.js]: Because I'm still used to c++ parameter rules I guess?! Removed, thanks.

**Review by willeastcott — APPROVED** [review]

**Pull request merged**